packages/api/src/controllers/webhook.ts

```diff
@@ -188,6 +196,48 @@ app.put("/:id", authorizer({}), validatePost("webhook"), async (req, res) => {
   res.json({ id: req.body.id });
 });
 
+app.patch(
+  "/:id",
+  authorizer({}),
+  validatePost("webhook-patch-payload"),
+  async (req, res) => {
+    const webhook = await db.webhook.get(req.params.id);
+
+    if (!webhook) {
+      throw new NotFoundError(`webhook not found`);
+    }
+
+    if (
+      (webhook.userId !== req.user.id || webhook.deleted) &&
+      !req.user.admin
+    ) {
+      // do not reveal that webhooks exists
+      throw new NotFoundError(`webhook not found`);
+    }
+
+    const { id, userId, createdAt, kind } = webhook;
+
+    if (req.body.streamId) {
+      const stream = await db.stream.get(req.body.streamId);
+      if (!stream) {
+        throw new NotFoundError(`stream not found`);
+      }
+    }
+
+    await db.webhook.update(req.params.id, {
+      ...req.body,
+      // non editable fields
+      id,
+      userId,
+      createdAt,
+      kind,
+    });
+
+    res.status(200);
+    res.json({ id: req.params.id });
```

Comment:
If we only return input values, might as well return a `204 No Content` instead. Could fetch the updated webhook to return it here but I don't think it's necessary (if you do do that tho, remember to clean write only fields).
```suggestion
    res.status(204).end();
```